### DISCUSSION TITLE
Can I skip using entrypoints, but specify some `index.html` for extension `options_page`

### DISCUSSION THREAD

**Discussion by honwhy** [wxt-dev/wxt#1057] [General]
ask for help.
Can I skip using entrypoints, but specify some `index.html` for extension `options_page`.
I am trying to transform a web project into extension project, and so the web entry index.html into options_page html.

However, as I can see wxt configuration definition was not support to specify `options_page`
```
//wxt.config.ts
export default defineConfig({
  manifest: {} // UserManifest
})
```

```
// definition of UserManifest which had omit options_page key
type UserManifest = Partial<Omit<Manifest.WebExtensionManifest, 'background' | 'chrome_url_overrides' | 'devtools_page' | 'manifest_version' | 'options_page' | 'options_ui' | 'sandbox'>>;

```

**Comment by aklinker1** [comment]
You can use the `build:manifestGenerated` hook to transform the manifest however you like. See https://wxt.dev/guide/key-concepts/manifest.html#full-control

Second code block shows you how to setup the hook in your `wxt.config.ts` file.

**Reply by honwhy** [reply]
There should be another issue, and rising another question, how to use wxt hook and wxt api to copy specify file such as `index.html` to `.output/chrome-mv3` directory for example.

This was my try. but the output directory does not exists index.html
```
  hooks: {
    'build:manifestGenerated': (wxt, manifest) => {
      manifest.options_page = 'index.html'
    },
  },
```

**Reply by aiktb** [reply]
> There should be another issue, and rising another question, how to use wxt hook and wxt api to copy specify file such as `index.html` to `.output/chrome-mv3` directory for example.
> 
> This was my try. but the output directory does not exists index.html
> 
> ```
>   hooks: {
>     'build:manifestGenerated': (wxt, manifest) => {
>       manifest.options_page = 'index.html'
>     },
>   },
> ```

```ts
  hooks: {
    "build:publicAssets": ({ config }, publicFiles) => {
      const filename = path.resolve(__dirname, "path/to/index.html");
      ...
      ...
      publicFiles.push({ absoluteSrc, relativeDest });
    },
  },
```
